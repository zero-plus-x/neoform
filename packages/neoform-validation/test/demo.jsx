/* eslint-disable max-len */
import React from 'react';
import { mount } from 'enzyme';

import Demo from '../demo';

describe('neoform-validation', () => {
  describe('field', () => {
    it('invalid', (done) => {
      const wrapper = mount(
        <Demo/>
      );

      const testField = wrapper
        .find('MyInput')
        .filterWhere((node) => node.prop('name') === 'friends.1.firstName');

      testField.prop('validate')();

      global.setImmediate(() => {
        expect(wrapper.find('MyForm').props()).toMatchSnapshot();
        expect(testField.props()).toMatchSnapshot();
        done();
      });
    });

    it('valid', (done) => {
      const wrapper = mount(
        <Demo/>
      );

      const testField = wrapper
        .find('MyInput')
        .filterWhere((node) => node.prop('name') === 'friends.1.firstName');

      testField.prop('onChange')('hey from test');
      testField.prop('validate')();

      global.setImmediate(() => {
        expect(wrapper.find('MyForm').props()).toMatchSnapshot();
        expect(testField.props()).toMatchSnapshot();
        done();
      });
    });

    it('without validator', (done) => {
      const wrapper = mount(
        <Demo/>
      );

      const testField = wrapper
        .find('MyInput')
        .filterWhere((node) => node.prop('name') === 'friends.1.lastName');

      testField.prop('validate')();

      global.setImmediate(() => {
        expect(wrapper.find('MyForm').props()).toMatchSnapshot();
        expect(testField.props()).toMatchSnapshot();
        done();
      });
    });

    it('unmount invalid field', (done) => {
      const wrapper = mount(
        <Demo/>
      );

      const testField = wrapper
        .find('MyInput')
        .filterWhere((node) => node.prop('name') === 'friends.1.firstName');

      testField.prop('validate')();

      global.setImmediate(() => {
        wrapper.setState({
          friends: [
            {
              firstName: 'Pepe',
              lastName: 'Sad'
            }
          ]
        }, () => {
          global.setTimeout(() => {
            expect(wrapper.find('MyForm').props()).toMatchSnapshot();
            done();
          }, 100);
        });
      });
    });
  });

  describe('form', () => {
    it('invalid', (done) => {
      const mockOnSuccess = jest.fn();
      const mockOnError = jest.fn();
      const wrapper = mount(
        <Demo/>
      );
      const form = wrapper.find('MyForm');

      form.prop('validate')(mockOnSuccess, mockOnError);

      global.setImmediate(() => {
        const testField = wrapper
          .find('MyInput')
          .filterWhere((node) => node.prop('name') === 'friends.1.firstName');

        expect(testField.props()).toMatchSnapshot();
        expect(form.props()).toMatchSnapshot();
        expect(mockOnSuccess).toHaveBeenCalledTimes(0);
        expect(mockOnError).toHaveBeenCalledTimes(1);
        done();
      });
    });

    it('valid', (done) => {
      const mockOnSuccess = jest.fn();
      const mockOnError = jest.fn();
      const wrapper = mount(
        <Demo/>
      );
      const form = wrapper.find('MyForm');
      const testField = wrapper
        .find('MyInput')
        .filterWhere((node) => node.prop('name') === 'friends.1.firstName');

      testField.prop('onChange')('hey from test');
      form.prop('validate')(mockOnSuccess, mockOnError);

      global.setImmediate(() => {
        expect(testField.props()).toMatchSnapshot();
        expect(form.props()).toMatchSnapshot();
        expect(mockOnSuccess).toHaveBeenCalledTimes(1);
        expect(mockOnError).toHaveBeenCalledTimes(0);
        done();
      });
    });
  });
});
