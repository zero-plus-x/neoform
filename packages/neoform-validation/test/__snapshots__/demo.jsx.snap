// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`neoform-validation field invalid 1`] = `
Object {
  "data": Object {
    "firstName": "John",
    "friends": Array [
      Object {
        "firstName": "Pepe",
        "lastName": "Sad",
      },
      Object {
        "firstName": "",
        "lastName": "Darkness",
      },
    ],
    "lastName": "Doe",
  },
  "getValue": [Function],
  "onChange": [Function],
  "onInvalid": [Function],
  "onSubmit": [Function],
  "validate": [Function],
  "validationStatus": false,
}
`;

exports[`neoform-validation field invalid 2`] = `
Object {
  "name": "friends.1.firstName",
  "onChange": [Function],
  "validate": [Function],
  "validationMessage": "required",
  "validationStatus": false,
  "value": "",
}
`;

exports[`neoform-validation field unmount invalid field 1`] = `
Object {
  "data": Object {
    "firstName": "John",
    "friends": Array [
      Object {
        "firstName": "Pepe",
        "lastName": "Sad",
      },
    ],
    "lastName": "Doe",
  },
  "getValue": [Function],
  "onChange": [Function],
  "onInvalid": [Function],
  "onSubmit": [Function],
  "validate": [Function],
  "validationStatus": true,
}
`;

exports[`neoform-validation field valid 1`] = `
Object {
  "data": Object {
    "firstName": "John",
    "friends": Array [
      Object {
        "firstName": "Pepe",
        "lastName": "Sad",
      },
      Object {
        "firstName": "hey from test",
        "lastName": "Darkness",
      },
    ],
    "lastName": "Doe",
  },
  "getValue": [Function],
  "onChange": [Function],
  "onInvalid": [Function],
  "onSubmit": [Function],
  "validate": [Function],
  "validationStatus": true,
}
`;

exports[`neoform-validation field valid 2`] = `
Object {
  "name": "friends.1.firstName",
  "onChange": [Function],
  "validate": [Function],
  "validationMessage": "valid",
  "validationStatus": true,
  "value": "hey from test",
}
`;

exports[`neoform-validation field without validator 1`] = `
Object {
  "data": Object {
    "firstName": "John",
    "friends": Array [
      Object {
        "firstName": "Pepe",
        "lastName": "Sad",
      },
      Object {
        "firstName": "",
        "lastName": "Darkness",
      },
    ],
    "lastName": "Doe",
  },
  "getValue": [Function],
  "onChange": [Function],
  "onInvalid": [Function],
  "onSubmit": [Function],
  "validate": [Function],
  "validationStatus": true,
}
`;

exports[`neoform-validation field without validator 2`] = `
Object {
  "name": "friends.1.lastName",
  "onChange": [Function],
  "validate": [Function],
  "validationMessage": undefined,
  "validationStatus": undefined,
  "value": "Darkness",
}
`;

exports[`neoform-validation form invalid 1`] = `
Object {
  "name": "friends.1.firstName",
  "onChange": [Function],
  "validate": [Function],
  "validationMessage": "required",
  "validationStatus": false,
  "value": "",
}
`;

exports[`neoform-validation form invalid 2`] = `
Object {
  "data": Object {
    "firstName": "John",
    "friends": Array [
      Object {
        "firstName": "Pepe",
        "lastName": "Sad",
      },
      Object {
        "firstName": "",
        "lastName": "Darkness",
      },
    ],
    "lastName": "Doe",
  },
  "getValue": [Function],
  "onChange": [Function],
  "onInvalid": [Function],
  "onSubmit": [Function],
  "validate": [Function],
  "validationStatus": false,
}
`;

exports[`neoform-validation form valid 1`] = `
Object {
  "name": "friends.1.firstName",
  "onChange": [Function],
  "validate": [Function],
  "validationMessage": "valid",
  "validationStatus": true,
  "value": "hey from test",
}
`;

exports[`neoform-validation form valid 2`] = `
Object {
  "data": Object {
    "firstName": "John",
    "friends": Array [
      Object {
        "firstName": "Pepe",
        "lastName": "Sad",
      },
      Object {
        "firstName": "hey from test",
        "lastName": "Darkness",
      },
    ],
    "lastName": "Doe",
  },
  "getValue": [Function],
  "onChange": [Function],
  "onInvalid": [Function],
  "onSubmit": [Function],
  "validate": [Function],
  "validationStatus": true,
}
`;
